broadinstitute/single_cell_portal_core
# Remove obsolete code references, prevent upload failure emails for deleted files (SCP-4690, SCP-4736)

#### BACKGROUND & CHANGES
This update fixes two small bugs:
* Removes a broken reference to `URI.escape` to restore the "Storage costs" feature in "My Billing Projects"
* Prevent sending "failed upload" emails for files that are queued for deletion

In addition, it removes some obsolete code for syncing files that is no longer referenced anywhere.

#### MANUAL TESTING
1. For the billing project bugfix, boot as normal and sign in, then select "My Billing Projects" from the profile menu
2. Next to your personal Terra billing project (e.g. `scp-development-{username}`), click the "Storage costs" button
3. Confirm the page loads and shows a breakdown of storage costs for workspaces
4. For the upload failure email bugfix, open a Rails console session and simulate a failed upload:
```
# first delete any queued files
StudyFile.delete_queued_files

# create a failed upload
study = Study.last
file = study.study_files.build(file_type: 'Other', upload: File.open(Rails.root.join('test/test_data/mock_study_doc_upload.txt')))
file.created_at = DateTime.now - 1.week
file.save
DeleteQueueJob.new(file).perform
```
5. Confirm the file is queued for deletion
```
file.reload
file.queued_for_deletion
=> true
```
6. Run the job to find and remove failed uploads, and confirm you do not get an email for `mock_study_doc_upload.txt` (you will see `ERROR -- : Error - #<Google::Apis::ClientError: notFound: No such object` in the console - this is expected as the file does not exist in the bucket)
```
UploadCleanupJob.find_and_remove_failed_uploads
```
